eng/pipelines/pull_request.yml

```diff
@@ -70,9 +70,39 @@ parameters:
   displayName: Run Mono tests on Mac
   type: boolean
   default: true
+- name: isOfficialBuild
+  type: boolean
+  default: false
+
+resources:
+  pipelines:
+  - pipeline: DartLab
+    source: DartLab
+    branch: main
+  repositories:
+  - repository: DartLabTemplates
+    type: git
+    name: DartLab.Templates
+    ref: refs/heads/main
```

Comment:
This YAML code is duplicated in three files in this PR. Can we consolidate this code into a single location and reference it as needed across all three places? The reason behind this request is that sometimes we may need to change the DartLab branch name to test if a fix on their side unblocks us. Generally, we avoid creating a pull request solely for changing the branch name. Instead, we queue a PR build to verify if the fix on their side resolves our issue, and then we revert our changes. DartLab team pushes their changes to the main branch eventually, which we rely on.

Reply:
That's a great thought, but unfortunately resources are not supported in YAML templates:
https://developercommunity.visualstudio.com/t/unexpected-value-resources-in-yaml-template/728151

If you know another way to accomplish this, do let me know.